### ISSUE TITLE
NetBox WebDAV & Mail.Ru Cloud

### ISSUE THREAD

**Issue by pasha-zzz** [michaellukashov/Far-NetBox#217]
Есть такая штука - мост WebDAV <-> Mail.Ru Cloud - https://github.com/yar229/WebDavMailRuCloud
И с NetBox есть проблема у него:

- NetBox пытается отправить большой файл (ограничение на бесплатных акках майл.ру - около 2 гиг)
- если файл больше максимально разрешенного размера, мост возвращает ошибку WebDAV "412 Precondition failed"
- после этого NetBox полностью кеширует файл в памяти и посылает его снова
- снова получает ошибку "412 Precondition failed"
- и только после этого NetBox выдает ошибку.

Клиент WebDAV Windows поступает так же, а Bitkinex работает как надо.

**Comment by VictorVG** [comment]
Так собрать-проверить не на чем - проблемы со сборкой и не понятно где их причины... https://github.com/michaellukashov/Far-NetBox/issues/216

**Comment by pasha-zzz** [comment]
ну так есть уже собранные...

**Comment by VictorVG** [comment]
Да, только они собраны в МС++2015 Upd 3 и будут работать на семёрке, а у ребят с ХР по их словам даже не запускаются. Потому и делаю для них сборку в 2010.

**Comment by pasha-zzz** [comment]
Для начала попробовать на том что есть - проблема на 7-10 точно присутствует. Ну а потом уже можно бекпортами заняться.

**Comment by VictorVG** [comment]
А у меня никак не выходит сборка. Похоже переменные среды компилятора не стоят. :?

**Comment by VictorVG** [comment]
Я тут у себя глянул ЯД на 2.4.2.442 уже своей сборки - задал логин-пароль и нет проблем, а ребята мучаются подбирая настройки. Может и тут похожие дела - подобрать настройки?

**Comment by pasha-zzz** [comment]
Нет. Не в настройках дело. Проблема в том, что НетБокс реагирует на ошибку со 2 раза. Причем после первого он пытается закешировать файл целиком и переотправить. А если файл немаленький - это много времени.

**Comment by VictorVG** [comment]
Ну, тогда смотреть надо.

**Comment by VictorVG** [comment]
С ЯД есть свои "тараканы" - там надо отключать кэширование и запоминание последнего каталога или кэширование "запирает" нас в последнем каталоге фиксируя его в настройках как корневой каталог сервера. У себя я это безобразие отключил:

```
<?xml version="1.0" encoding="UTF-8"?>
<NetBox version="2.1">
    <Sessions>
        <Session name="MYLOGIN@webdav.yandex.ru">
            <Version>2.4.2</Version>
            <HostName>webdav.yandex.ru</HostName>
            <PortNumber>443</PortNumber>
            <UserName>MYLOGIN</UserName>
            <FSProtocol>WebDAV</FSProtocol>
            <RemoteDirectory>/</RemoteDirectory>
            <UpdateDirectories>No</UpdateDirectories>
            <CacheDirectories>No</CacheDirectories>
            <CacheDirectoryChanges>No</CacheDirectoryChanges>
            <LookupUserGroups2>0</LookupUserGroups2>
            <SFTPMaxVersion>0</SFTPMaxVersion>
            <Ftps>2</Ftps>
            <CodePage></CodePage>
            <LoginType>1</LoginType>
            <Password>MYPASSWORD</Password>
        </Session>
    </Sessions>
</NetBox>
```
Пароль и логин по месту.
